### ISSUE TITLE
Feature request: Apply adapter bid adjustment to unknown bidder codes

### ISSUE THREAD

**Issue by patmmccann** [prebid/Prebid.js#8312]
https://github.com/prebid/Prebid.js/pull/8216 carved out applying the adapter code bid adjustment function to unknown bidder codes. We would still like that feature, but since it is breaking, we would want to put it behind configuration. No known bidders currently return 'surprise' bidder codes (april 2022), but we would like to make sure adjustment functions are able to handle them should that occur.

**Comment by jlquaccia** [comment]
Hey @patmmccann, was taking a look into this one but had a question about the ask.  Say we have the following bidderSettings in place:

```
        pbjs.bidderSettings = {
          standard: {
            bidCpmAdjustment: function (bidCpm, bid) {
              return bidCpm * 0.95;
            },
          },
          pubmatic: {
            catchUnknownBidderCodesWithAdapterBidAdjustment: true, ( new config option )
            bidCpmAdjustment: function (bidCpm, bid) {
              return bidCpm * 0.9;
            }
          }
        };
```

If the PubMatic response returned a bidder code of "unknownCode" then that bidderCode would normally get rejected since by default `allowAlternateBidderCodes` is set to false for `standard` and `pubmatic` above.  Would the ask for this ticket then, in this scenario, be to allow the unknown code to pass through and use the `pubmatic.bidCpmAdjustment` function? (since this unknown code came from pubmatic and the new config option above is present and set to allow this).

Basically, the `allowAlternateBidderCodes` and `allowedAlternateBidderCodes` functionality would override `catchUnknownBidderCodesWithAdapterBidAdjustment`, but `catchUnknownBidderCodesWithAdapterBidAdjustment` could optionally be used as a fallback if the other keys were absent, instead of just rejecting all bids with unknown bidder codes.
